storm-core/src/jvm/org/apache/storm/bolt/JoinBolt.java

```diff
@@ -191,9 +191,9 @@ public void execute(TupleWindow inputWindow) {
         for (ResultRecord resultRecord : joinResult.getRecords()) {
             ArrayList<Object> outputTuple = resultRecord.getOutputFields();
             if ( outputStreamName==null )
-                collector.emit( outputTuple );
+                collector.emit( resultRecord.tupleList, outputTuple );
```

Comment:
It may be good to add a comment here, saying that anchoring with only matched input tuples as .
`WindowedOutputCollector` anchors with all the incoming tuples.

For others to understand the issue, collector used here is [`WindowedOutputCollector`](https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/topology/WindowedBoltExecutor.java#L31) and it emits  output tuple by anchoring with all the input tuples with in that window.